mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -118,9 +118,11 @@ private[spark] object RandomForest extends Logging {
     val splits = findSplits(retaggedInput, metadata, seed)
     timer.stop("findSplits")
     logDebug("numBins: feature: number of bins")
-    logDebug(Range(0, metadata.numFeatures).map { featureIndex =>
-      s"\t$featureIndex\t${metadata.numBins(featureIndex)}"
-    }.mkString("\n"))
+    if (log.isDebugEnabled) {
+      logDebug(Range(0, metadata.numFeatures).map { featureIndex =>
```

Comment:
Now if we have a very big vector (likes 100k and more dimension) it may take a few minutes to don't print a debug log.